syntax = "proto3";

package users;

import "google/api/annotations.proto";


service UsersService {
	rpc UserExists (UserExistsRequest) returns (UserExistsResponse) {
	  option (google.api.http) = {
		get: "/v1/users/{userLogin}"
	  };
	}
	rpc CreateUser (CreateUserRequest) returns (CreateUserResponse) {
		option (google.api.http) = {
			post: "/v1/users/create",
			body: "*" // В body весь CreateUserRequest
		};
	}
}

message UserExistsRequest {
	string userLogin = 1;
}
message UserExistsResponse {
	bool exists = 1;
}

message CreateUserRequest {
	string userLogin = 1;
}
message CreateUserResponse {
	string message = 1;
}
